# DistRMI

This is a Pytorch implementation of the paper:"DistRMI: a deep distance-aware neural network for explainable RNA loop motif-small molecule interaction prediction"

![framework]![framework](figures/framework.png)



# Requirements

The model was developed and tested in a Linux environment (Ubuntu 22.04.5 LTS) using Python 3.11 and a CUDA-enabled GPU, NVIDIA GeForce RTX 3090, with CUDA version 12.4 and driver version 550.127.05.

* Python >= 3.11
* PyTorch >= 1.0
* RDkit >= 2024.09
* scikit-learn >=1.5

You can install the dependencies via:

```
pip install -r requirements.txt
```

or using the optional conda environment:

```
conda env create -f environment.yml
conda activate distRMI
```



# Usage

**Training**

Training the model by running

```
python training_validation.py
```

Hyperparameters

| Hyperparameter                   | Optimal Value | Description                                             |
| -------------------------------- | ------------- | ------------------------------------------------------- |
| Dropout                          | 0.4           | Dropout rate applied to prevent overfitting             |
| Embedding layer dimension (k)    | 128           | Dimension of k-mer embeddings                           |
| Weight fusion ratio α            | 0.1           | Balances basic attention and molecule-distance features |
| Auxiliary loss ratio β           | 0.3           | Balances auxiliary loss terms                           |
| Number of Transformer layers (l) | 2             | Depth of Transformer encoder                            |
| Learning rate (lr)               | 0.001         | Initial learning rate                                   |
| Number of attention heads (n)    | 16            | Multi-head attention heads                              |
| Epochs                           | 30            | Total training epochs                                   |

Input:

* RNA loop motif sequences ( .xlsx  files) 
  - Loop_Data_ID: Unique identifier for each RNA sequence
  - 1D Sequence: Nucleotide sequence of the loop motif
* Small molecule SMILES strings (.xlsx  files)
  - Ligand_Data_ID: Unique identifier for each molecule
  - CanonicalSMILES: SMILES string
* RNA loop motif-Small molecule binary interaction labels (0/1) (.xlsx  files)
  - Loop_Data_ID: Unique identifier for each RNA sequence
  - Ligand_Data_ID: Unique identifier for each molecule
  - Label: 0 = no binding, 1 = binding
* site_distance (.xlsx  files) – contains distances between RNA nucleotides and small molecule binding sites
  - Ligand_Data_ID: Unique identifier for each molecule
  - Loop_Data_ID: Unique identifier for each RNA sequence
  - New base ID: Unique identifier for each RNA Nucleotide
  - Distance: Distance (Å) from the RNA nucleotide to the small molecule binding site

Output:

* Trained model weights (.pt files)

**Prediction**

Predictions from the models can be generated by running

```
python predict.py
```

Input:

- Prediction input (.xlsx file)
  - Loop_Data_ID: Unique identifier for each RNA sequence
  - 1D Sequence: Nucleotide sequence of the loop motif
  - Ligand_Data_ID: Unique identifier for each molecule
  - CanonicalSMILES: SMILES string


Output:

- Predicted interaction labels (.xlsx file)
  - Loop_Data_ID: Unique identifier for each RNA sequence
  - 1D Sequence: Nucleotide sequence of the loop motif
  - Ligand_Data_ID: Unique identifier for each molecule
  - CanonicalSMILES: SMILES string
  - Predicted interaction labels (0 = no binding, 1 = binding)
